test/no-array-instanceof.js

```diff
@@ -31,38 +31,23 @@ ruleTester.run('avoid-array-instanceof', rule, {
 			errors
 		},
 		{
-			code: 'arr instanceof Array === true',
-			output: 'Array.isArray(arr) === true',
-			errors
-		},
-		{
 			code: '[] instanceof Array',
 			output: 'Array.isArray([])',
 			errors
 		},
 		{
-			code: '[] instanceof Array === true',
-			output: 'Array.isArray([]) === true',
-			errors
-		},
-		{
-			code: '[1,2,3] instanceof Array',
-			output: 'Array.isArray([1,2,3])',
-			errors
-		},
-		{
-			code: '[1,2,3] instanceof Array === true',
-			output: 'Array.isArray([1,2,3]) === true',
+			code: 'fun.call(1, 2, 3) instanceof Array',
+			output: 'Array.isArray(fun.call(1, 2, 3))',
 			errors
 		},
 		{
-			code: 'if(arr instanceof Array){}',
-			output: 'if(Array.isArray(arr)){}',
+			code: 'obj.arr instanceof Array',
+			output: 'Array.isArray(obj.arr)',
 			errors
 		},
 		{
-			code: 'if([1,2,3] instanceof Array){}',
-			output: 'if(Array.isArray([1,2,3])){}',
+			code: `foo.bar${value} instanceof Array`,
```

Comment:
@jfmengels I wanted to see how the rule would handle this case. I tried the same template literal in AST explorer and it doesn't evaluate to BinaryExpression at all. All it does is parse it as TemplateLiteral. So, I am currently confused how the tests are passing. :(

[AST Explorer Sample](https://astexplorer.net/#/hyq1piy29D)